src/modules/job-info/job_util.c

```diff
@@ -120,7 +120,11 @@ json_t *job_to_json (struct job *job, json_t *attrs, job_info_error_t *errp)
         else if (!strcmp (attr, "ranks")) {
             if (!(job->states_mask & FLUX_JOB_RUN))
                 continue;
-            val = json_string (job->ranks);
```

Comment:
Suggest you break the inner block of the loop out to its own function as this is getting pretty long and the control flow harder to follow.

Reply:
since this is a cleanup not so much related to this PR, i'll leave this one for another day given our 0.19.0 push